### ISSUE TITLE
Handling multiple requests

### ISSUE THREAD

**Issue by nyurik** [bjoerge/debounce-promise#6]
I would like to optimize multiple expensive server calls with the following logic:
* on first call, create a promise
* on each call, accumulate data to be sent to the server
* when no calls are made for 100 ms, call the server to process all the requests accumulated so far
* if a new call is made (even if the server hasn't responded yet), treat it as the "first call" - starting a new promise

Can I do something like this with your lib?  Thanks!

**Comment by bjoerge** [comment]
This is an interesting use case. Currently it is not possible, but I think it could be covered by an `accumulate` option, that accumulates all arguments passed while the debounced function is in the pending state.

```js
function fetchBatch(ids) {
  return fetch(`/resource?ids=${ids.join(',')}`)
}

const fetchById = debounce(fetchBatch, 100, {accumulate: true})

fetchById(1).then(resultOf1 => {
  
})
fetchById(2).then(resultOf2 => {

})
// ...
```

Some quick thoughts:
We need to map arguments back to promise values, this could be done with an `argument position` => `promise` index.

But then, what about multiple equal arguments, e.g. `fetchById(1); fetchById(1)`? We could keep a hash of `arg => promise`, but that becomes messy when arguments are non-primitives. (not too keen on using WeakMap that requires polyfill in older envs).

**Comment by nyurik** [comment]
I started a discussion on stackoverflow - there is an anwsere there, i'm still evaluating it
http://stackoverflow.com/questions/39970101/combine-multiple-debounce-promises-in-js - I outlined some corner cases here. Multiple identical requests should simply be ignored - they would return the same promise anyway.

**Comment by bjoerge** [comment]
The `accumulate` option is implemented in 2.1.0 by doing an `argument position` => `promise` mapping. Check out the usage example in readme.
